gridstatus/spp.py

```diff
@@ -855,18 +860,29 @@ def _get_rtm5_lmp(
         self,
         date,
         end=None,
+        location_type=LOCATION_TYPE_BUS,
         verbose=False,
     ):
+        fs_name = (
+            RTBM_LMP_BY_BUS
+            if location_type == LOCATION_TYPE_BUS
+            else FS_RTBM_LMP_BY_LOCATION
+        )
+
+        latest_file_id = (
+            "RTBM-LMP-B" if location_type == LOCATION_TYPE_BUS else "RTBM-LMP-SL"
+        )
+
         if date == "latest":
             urls = [
                 FILE_BROWSER_DOWNLOAD_URL
                 + "/"
-                + FS_RTBM_LMP_BY_LOCATION
-                + "?path=%2FRTBM-LMP-SL-latestInterval.csv",
+                + fs_name
+                + f"?path=%2F{latest_file_id}-latestInterval.csv",
             ]
         else:
             urls = self._file_browser_list(
```

Comment:
a big downside to our current approach here of using `_file_browser_list` is it only supports querying for one day at a time.

the files we are downloading have a known url structure, so there is no need to this operation where we list things. we can just directly construct the URL for the interval we want to load. 

this approach is currently taken by [`get_lmp_real_time_weis`](https://github.com/kmax12/gridstatus/blob/affd7e8d306c5a8f65b784dc2069a3e0a3fdda5d/gridstatus/spp.py#L1055)

we should update this function to look more like that and refactor this to not use `_file_browser_list`. 

in doing this, i think it make sense to break apart the current `get_lmp` function in the three functions for each of separate datasets. this will break our current api, but i think we should eventually move this library away from having the `get_lmp` function that takes `market` or `location_type` as a parameter to decide which dataset to grab

Reply:
I've updated the PR to separate the `get_lmp` method into three functions:

* `get_lmp_real_time_5_min_by_location`
* `get_lmp_real_time_5_min_by_bus`
* `get_lmp_day_ahead_hourly`

I've also removed `_file_browser_list` in favor of constructing the URLs through string interpolation.